planner/core/expression_rewriter.go

```diff
@@ -843,6 +843,22 @@ func (er *expressionRewriter) Leave(originInNode ast.Node) (retNode ast.Node, ok
 		er.isTrueToScalarFunc(v)
 	case *ast.DefaultExpr:
 		er.evalDefaultExpr(v)
+	// TODO: Perhaps we don't need to transcode these back to generic integers/strings
+	case *ast.TrimDirectionExpr:
+		er.ctxStack = append(er.ctxStack, &expression.Constant{
+			Value:   types.NewIntDatum(int64(v.Direction)),
+			RetType: types.NewFieldType(mysql.TypeTiny),
+		})
+	case *ast.TimeUnitExpr:
```

Comment:
@tiancaiamao Do we need to port these changes to simple_expression_rewriter?

Reply:
Yes, here: https://github.com/pingcap/tidb/blob/master/expression/simple_rewriter.go#L151
Thanks to @coocood 's great job!